### ISSUE TITLE
BUG - Modifying tags works but may throw an exception - fix included

### ISSUE THREAD

**Issue by nickion** [mautic/mautic#3109]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? | Y
| Feature request? | N
| Enhancement? | N

## Description:
Adding tags via the PHP API 2.4 where the tags already exist and the lead already has those tags causes the operation to fail. 

The error response would be something such as:
```
Array
(
    [error] => Array
        (
            [message] => Catchable Fatal Error: Argument 1 passed to Mautic\LeadBundle\Entity\Lead::addTag() must be an instance of Mautic\LeadBundle\Entity\Tag, null given, called in /var/www/html/mautic-2.4.0/app/bundles/LeadBundle/Model/LeadModel.php on line 1929 and defined
            [code] => 500
        )
)
```

## If a bug:

| Q   | A
| --- | ---
| Mautic version | 2.4
| PHP version | 5.6

### Steps to reproduce:
1. Add tags to a lead where they already have the tags
 
The following code for modifyTags will resolve. Adjust as you wish.

```
$tagtobeAdded = null; // ensure initialised and reset on each iteration
if (!array_key_exists($tag, $foundTags)) {
                    $tagToBeAdded = new Tag();
                    $tagToBeAdded->setTag($tag);
} elseif (!$leadTags->contains($foundTags[$tag])) {
                    $tagToBeAdded = $foundTags[$tag];
}

// $tagtoBeAdded could be null so we need to test it
if ($tagtobeAdded) {
                $lead->addTag($tagToBeAdded);
                $tagsModified = true;

                $this->logger->debug('CONTACT: Added '.$tag);
}
```

**Comment by escopecz** [comment]
Fixed in https://github.com/mautic/mautic/pull/3101

**Comment by nickion** [comment]
Thanks John. I've commented on the changes as they are still faulty (as was my suggestion here from case typos unfortunately). In the fix as it stands, the same tag may be added multiple times from $tagToBeAdded not being cleared on each iteration.

**Comment by escopecz** [comment]
@nickion would you be able to send a pull request?

**Comment by virlatinus** [comment]
This is already fixed in 2.7
